packages/playground/blueprints/src/lib/steps/install-plugin.ts

```diff
@@ -58,110 +59,152 @@ export const installPlugin: StepHandler<InstallPluginStep<File>> = async (
 	{ pluginZipFile, options = {} },
 	progress?
 ) => {
-	progress?.tracker.setCaption(
-		`Installing the ${zipNameToHumanName(pluginZipFile?.name)} plugin`
-	);
+	const zipFileName = pluginZipFile.name.split('/').pop() || 'plugin.zip';
+	const zipNiceName = zipNameToHumanName(zipFileName);
+
+	progress?.tracker.setCaption(`Installing the ${zipNiceName} plugin`);
 	try {
-		const activate = 'activate' in options ? options.activate : true;
+		// Extract to temporary folder so we can find plugin folder name
+
+		const tmpFolder = '/tmp/plugin';
+		const tmpZipPath = `/tmp/${zipFileName}`;
 
-		// Upload it to WordPress
-		const pluginForm = await playground.request({
-			url: '/wp-admin/plugin-install.php?tab=upload',
+		if (await playground.isDir(tmpFolder)) {
+			await playground.unlink(tmpFolder);
+		}
+
+		await writeFile(playground, {
+			path: tmpZipPath,
+			data: pluginZipFile,
 		});
-		const pluginFormPage = asDOM(pluginForm);
-		const pluginFormData = new FormData(
-			pluginFormPage.querySelector('.wp-upload-form')! as HTMLFormElement
-		) as any;
-		// eslint-disable-next-line @typescript-eslint/no-unused-vars
-		const { pluginzip, ...postData } = Object.fromEntries(
-			pluginFormData.entries()
-		);
 
-		const pluginInstalledResponse = await playground.request({
-			url: '/wp-admin/update.php?action=upload-plugin',
-			method: 'POST',
-			formData: postData,
-			files: { pluginzip: pluginZipFile },
+		await unzip(playground, {
+			zipPath: tmpZipPath,
+			extractToPath: tmpFolder,
 		});
 
-		// Activate if needed
-		if (activate) {
-			const pluginInstalledPage = asDOM(pluginInstalledResponse);
-			const activateButtonHref = pluginInstalledPage
-				.querySelector('#wpbody-content .button.button-primary')!
-				.attributes.getNamedItem('href')!.value;
-			const activatePluginUrl = new URL(
-				activateButtonHref,
-				await playground.pathToInternalUrl('/wp-admin/')
-			).toString();
-			await playground.request({
-				url: activatePluginUrl,
-			});
+		await playground.unlink(tmpZipPath);
+
+		// Find extracted plugin folder name
+
+		const files = await playground.listFiles(tmpFolder);
+
+		let pluginFolderName;
+		let tmpPluginPath = '';
+
+		for (const file of files) {
+			tmpPluginPath = `${tmpFolder}/${file}`;
```

Comment:
This is such a common use-case it would be nice to have a `{ includePath: boolean }` option, or something to that effect, for `listFiles`.

Reply:
Started a PR draft.

- #462